AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -899,6 +890,51 @@ - (void)transitionContext:(_ASTransitionContext *)context didComplete:(BOOL)didC
   _transitionContext = nil;
 }
 
+#pragma mark - Layout
+
+- (void)_completePendingLayoutTransition
+{
+  ASDN::MutexLocker l(_propertyLock);
+  if (_pendingLayoutTransition) {
+    [self _setCalculatedLayout:_pendingLayoutTransition.pendingLayout];
+    [self _completeTransition:_pendingLayoutTransition];
+    _pendingLayoutTransition = nil;
+  }
+  [self _completeLayoutCalculation];
+}
+
+- (void)_setCalculatedLayout:(ASLayout *)layout
+{
+  ASDN::MutexLocker l(_propertyLock);
+  
+  ASDisplayNodeAssertTrue(layout.layoutableObject == self);
+  ASDisplayNodeAssertTrue(layout.size.width >= 0.0);
+  ASDisplayNodeAssertTrue(layout.size.height >= 0.0);
+  
+  _layout = layout;
+}
+
+- (void)_completeTransition:(ASLayoutTransition *)layoutTransition
+{
+  // Layout transition is not supported for non implicit hierarchy managed nodes
+  if (layoutTransition == nil || self.usesImplicitHierarchyManagement == NO) {
```

Comment:
Probably don't need to check `self.usesImplicitHierarchyManagement` anymore, since it's just a feature flag and you won't be able to get to this point unless you're invoking `transitionLayout` on the node.